app/api/mapping/permissions.py

```diff
@@ -39,6 +39,34 @@ def is_az_function_user(user: User) -> bool:
     return user.username == os.getenv("AZ_FUNCTION_USER")
 
 
+def is_scan_report_author(obj: Any, request: Request) -> bool:
+    """Check if the user is the author of a scan report.
+
+    Args:
+        obj (Any): The object to check (should be a ScanReport or related object).
+        request (Request): The request with the User instance.
+
+    Returns:
+        bool: `True` if the request's user is the author of the scan report, else `False`.
+    """
+    try:
+        # If `obj` is a scan report table|field|value, get the scan report
+        # it belongs to and check the user has permission to view it.
+        if sub_scan_report_query := SCAN_REPORT_QUERIES.get(type(obj)):
+            sub_scan_report = sub_scan_report_query(obj)
+        else:
+            sub_scan_report = obj
+
+        # Check if the user is the author
+        if isinstance(sub_scan_report, ScanReport):
+            is_author = sub_scan_report.author.id == request.user.id
+            return is_author
```

Comment:
```suggestion
            return sub_scan_report.author.id == request.user.id
```

Reply:
Aren't they the same? I thought return is_author would be also shorter. I don't fully understand. sorry

Reply:
So you're assigning the results of the comparison to a variable `is_author` and then immediately returning that variable. You may as well just return the comparison. It will do the same thing.

Reply:
Ah, I got what you mean when looking at the VS code, not on Github. Pushed the change. Thanks